lib/types/Cargo.toml

```diff
@@ -16,9 +16,18 @@ name = "wasmer_types"
 [dependencies]
 thiserror = "1.0"
 indexmap = { version = "1.6" }
+num-traits = "0.2.15"
```

Comment:
My intuition here would be to “just” hard code the `u32` key instead, over bringing in this dependency. OTOH, `num-traits` is already in the dep-graph so :shrug:.

Reply:
My thoughts are, with num-traits we get the same benefits as with the sealed trait previously, aka being able to open-source PartialSumMap as a split crate easily :)